src/Logary/LogResultModule.fs

```diff
@@ -0,0 +1,38 @@
+namespace Logary
+
+[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+module internal LogResult =
+  open Hopac
+
+  let success: LogResult = Alt.always (Result.Ok Promise.unit)
+  let error message : LogResult = Alt.always (Result.Error message)
+
+  let notAcceptByTarget (targetName: string) = error (Message.eventFormat("not accept by {target}", targetName))
+
+module internal LogError =
+  let targetBufferFull (targetName: string): ProcessResult = Result.Error (Message.eventFormat("{target} 's log buffer is full", targetName))
+
+  let timeOutToPutBuffer (targetName: string) (waitSeconds: double): ProcessResult = Result.Error (Message.eventFormat("time out: waiting {duration} seconds to putting message on to {target} 's log buffer", waitSeconds, targetName))
```

Comment:
This is very dangerous territory; we've designed the buffers to either return true or false; if they time out, it means the scheduler is overcrowded, which means we'll probably crash. We don't want to allocate then, nor will anything be available to actually log/write this message, since nothing gets scheduled anew.

Reply:
the logging error message return by the logging is not designed by resend to the buffer, it gives a chance to the client to notify the error details just like the old DU `LogError`, and users can choose a safe way to get the notification by the error handler.

Reply:
Can we at least make it non-allocating, besides the insertion of the variables into an object?